js_modules/dagster-ui/packages/ui-core/src/asset-data/AssetLiveDataProvider.tsx

```diff
@@ -1,172 +1,91 @@
-import {useApolloClient} from '@apollo/client';
+import {ApolloClient, gql, useApolloClient} from '@apollo/client';
 import uniq from 'lodash/uniq';
 import React from 'react';
 
-import {AssetDataRefreshButton} from './AssetDataRefreshButton';
-import {AssetLiveDataThreadID} from './AssetLiveDataThread';
-import {AssetLiveDataThreadManager} from './AssetLiveDataThreadManager';
+import {
+  AssetGraphLiveQuery,
+  AssetGraphLiveQueryVariables,
+} from './types/AssetLiveDataProvider.types';
+import {SUBSCRIPTION_MAX_POLL_RATE} from './util';
 import {observeAssetEventsInRuns} from '../asset-graph/AssetRunLogObserver';
-import {LiveDataForNode, tokenForAssetKey} from '../asset-graph/Utils';
+import {
+  LiveDataForNode,
+  buildLiveDataForNode,
+  tokenForAssetKey,
+  tokenToAssetKey,
+} from '../asset-graph/Utils';
 import {AssetKeyInput} from '../graphql/types';
-import {useDocumentVisibility} from '../hooks/useDocumentVisibility';
+import {liveDataFactory} from '../live-data-provider/Factory';
+import {LiveDataPollRateContext} from '../live-data-provider/LiveDataProvider';
+import {LiveDataThreadID} from '../live-data-provider/LiveDataThread';
 import {useDidLaunchEvent} from '../runs/RunUtils';
 
-export const SUBSCRIPTION_IDLE_POLL_RATE = 30 * 1000;
-const SUBSCRIPTION_MAX_POLL_RATE = 2 * 1000;
-
-export const LiveDataPollRateContext = React.createContext<number>(SUBSCRIPTION_IDLE_POLL_RATE);
-
-export const AssetLiveDataRefreshContext = React.createContext<{
-  isGloballyRefreshing: boolean;
-  oldestDataTimestamp: number;
-  refresh: () => void;
-}>({
-  isGloballyRefreshing: false,
-  oldestDataTimestamp: Infinity,
-  refresh: () => {},
-});
-
-export function useAssetLiveData(
-  assetKey: AssetKeyInput,
-  thread: AssetLiveDataThreadID = 'default',
-) {
-  const {liveDataByNode, refresh, refreshing} = useAssetsLiveData(
-    React.useMemo(() => [assetKey], [assetKey]),
-    thread,
+function makeFactory() {
+  return liveDataFactory(
+    () => {
+      return useApolloClient();
```

Comment:
Burying a hook down in here feels a bit gross because of how many rules surround use of hooks